src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -2477,3 +2484,353 @@ def _aks_mesh_update(
         return None
 
     return aks_update_decorator.update_mc(mc)
+
+
+MESH_CONFIG_ALLOW_LIST = {
+    '1.17.2': {
+        'defaultConfig',
+        'outboundTrafficPolicy',
+        'extensionProviders'
+        'configSources',
+        'defaultProviders',
+        'enableTracing',
+        'accessLogFile',
+        'accessLogFormat',
+        'accessLogEncoding',
+        'accessEnvoyAccessLogService',
+        'disablEnvoyListenerLog',
+        'caCertificates',
+        'enablePrometheusMerge',
+        'discoverySelectors',
+        'rootNamespace'
+    },
+    '1.18.0': {
+        'defaultConfig',
+        'outboundTrafficPolicy',
+        'extensionProviders'
+        'configSources',
+        'defaultProviders',
+        'enableTracing',
+        'accessLogFile',
+        'accessLogFormat',
+        'accessLogEncoding',
+        'accessEnvoyAccessLogService',
+        'disablEnvoyListenerLog',
+        'caCertificates',
+        'enablePrometheusMerge',
+        'discoverySelectors',
+        'rootNamespace'
+    }
+}
+
+
+PROXY_CONFIG_ALLOW_LIST = {
+    '1.17.2': {
+        'tracingServiceName',
+        'drainDuration',
+        'tracing',
+        'concurrency',
+        'envoyMetricsService',
+        'terminationDrainDuration',
+        'caCertificatesPem'
+    },
+    '1.18.0': {
+        'tracingServiceName',
+        'drainDuration',
+        'tracing',
+        'concurrency',
+        'envoyMetricsService',
+        'terminationDrainDuration',
+        'caCertificatesPem'
+    }
+
+}
+
+result_dict = {}
+
+
+def aks_mesh_migration_check(cmd, client, resource_group_name, name, kubeconfig = os.path.join(os.path.expanduser("~"), ".kube", "config")):
+    from azure.cli.core import __version__ as local_version
+    from packaging.version import parse
+    from pprint import pprint
+    
+    #Look for KUBECONFIG env variable in case of no argument from user
+    if "KUBECONFIG" in os.environ and kubeconfig == os.path.join(os.path.expanduser('~'), '.kube', 'config'):
+        kubeconfig_path = os.environ["KUBECONFIG"].split(os.pathsep)[0]
+        if kubeconfig_path:
+            logger.info("The default path '%s' is replaced by '%s' defined in KUBECONFIG.", kubeconfig, kubeconfig_path)
+            kubeconfig = kubeconfig_path
+        else:
+            logger.warning("Invalid path '%s' defined in KUBECONFIG.", kubeconfig_path)
+ 
+    contexts, _ = kubernetes.config.list_kube_config_contexts()
+    cur_context = load_kube_context(name, contexts)
+    
+    #ERROR_HANDLING
+    if (cur_context == ''):
+        raise CLIError("Couldn't load the kube context with the clustername provided. Please check cluster name or kubeconfig.")
+
+    kubernetes.config.load_kube_config(context = cur_context, config_file = kubeconfig)
+    v1 = kubernetes.client.CoreV1Api()
+    aks_mesh_check_base_requirements(cmd, client, resource_group_name, name, v1)
```

Comment:
I think the tool should return early if the add-on is already installed. It's a simple check quite similar to the one you have in `aks_mesh_check_istio_existence`:
```suggestion
    v1 = kubernetes.client.CoreV1Api()
    
    # if the service mesh add-on is already enabled, return without running checks
    asm_istiod = v1.list_namespaced_pod(namespace="aks-istio-system", label_selector="app=istiod")
    if len(asm_istiod.items) > 0:
       print("Azure service mesh add-on is already installed!")
       return
    aks_mesh_check_base_requirements(cmd, client, resource_group_name, name, v1)
```